BGAnimations/ScreenSelectMusic overlay/SongSearch/default.lua

```diff
@@ -0,0 +1,206 @@
+		end
+	},
+}
+
+local overlay = Def.ActorFrame {
+	Name="Overlay",
+	InitCommand=function(self)
+		self:xy(_screen.cx, _screen.cy + 40)
+	end,
+
+	AssessCandidatesCommand=function(self, params)
+		self:GetChild("SearchText"):playcommand("UpdateText", params)
+		self:GetChild("NumResults"):playcommand("UpdateText", params)
+		local candidates = {}
+		for candidate in ivalues(params.candidates) do
+			table.insert(candidates, {
+				index=#candidates,
+				songOrExit=candidate,
+				totalItems=#params.candidates + 1
+			})
+		end
+		table.insert(candidates, {
+			index=#candidates,
+			songOrExit="Exit",
+			totalItems=#params.candidates + 1
+		})
+		-- candidatesScroller.disable_wrapping = true
+		-- candidatesScroller.focus_pos = 1
+		candidatesScroller:set_info_set(candidates, 1)
+		self:playcommand("UpdateScrollbar",  {numCandidates = #params.candidates})
+	end,
+
+	-- White border
+	Def.Quad {
+		InitCommand=function(self)
+			self:diffuse(Color.White)
+			self:zoomto(paneWidth + borderWidth, paneHeight + borderWidth)
+		end,
+	},
+
+	-- Main black body
+	Def.Quad {
+		InitCommand=function(self)
+			self:diffuse(Color.Black)
+			self:zoomto(paneWidth, paneHeight)
+		end,
+	},
+
+	Def.Quad {
+		InitCommand=function(self)
+			self:diffuse(color("0.2,0.2,0.2"))
+			self:zoomto(borderWidth, paneHeight - 10)
+		end,
+	},
+
+	LoadFont("Common Normal").. {
+		Text="Search Results For:",
+		InitCommand=function(self)
+			self:diffuse(Color.White)
+			self:y(-paneHeight/2 - textHeight * 5)
+		end,
+	},
+
+	LoadFont("Common Normal").. {
+		Name="SearchText",
+		InitCommand=function(self)
+			self:diffuse(Color.White)
+			self:y(-paneHeight/2 - textHeight * 3)
+		end,
+		UpdateTextCommand=function(self, params)
+			self:settext("\""..params.searchText.."\"")
+			self:AddAttribute(1, {Length=#self:GetText()-2; Diffuse=Color.Green})
+		end,
+	},
+
+	LoadFont("Common Normal").. {
+		Name="NumResults",
+		InitCommand=function(self)
+			self:diffuse(Color.White)
+			self:maxwidth(paneWidth/2)
+			self:y(-paneHeight/2 - textHeight)
+		end,
+		UpdateTextCommand=function(self, params)
+			self:settext(#params.candidates.." Results Found")
+		end
+	},
+
+	candidatesScroller:create_actors("Candidates", 12, candidateItemMt, -paneWidth/4, -paneHeight/2 - textHeight * 2.5)
+}
+
+local songDetails = {
+	{ "Pack", function(song) return song:GetGroupName() end },
+	{ "Song", function(song) return song:GetDisplayMainTitle() end },
+	{ "Subtitle", function(song) return song:GetDisplaySubTitle() end },
+	{ "BPMs", function(song)
+		local bpms = song:GetDisplayBpms()
+		if bpms[2]-bpms[1] == 0 then
+			return string.format("%.0f", bpms[1])
+		else
+			return string.format("%.0f - %.0f", bpms[1], bpms[2])
+		end
+	end },
+	{ "Difficulties", function(song)
+		local stepsType = GAMESTATE:GetCurrentStyle():GetStepsType()
+		local difficulties = {
+			"Difficulty_Beginner",
+			"Difficulty_Easy",
+			"Difficulty_Medium",
+			"Difficulty_Hard",
+			"Difficulty_Challenge"
```

Comment:
What about edit difficulty?

Reply:
I specifically disregarded edit difficulties. I personally thought it was less useful and since there can be any number of them, displaying them all seemed unnecessary.

Reply:
Hm, I think we shouldn't special-case edit difficulty. I've never seen a chart with more than three, so I don't think that space is an issue here.

Reply:
I have /definitely/ seen charts with more than 3. There are a lot of examples even in the UPS packs where some have more than 10 (one proposed submission had 80+). I very much think that people do not actively seek out edit charts and I pretty strongly think it shouldn't be included.

Reply:
Also additionally, GetOneSteps will probably only really return the first edit which I think is also a pretty meh experience since it's like not quite deterministic (although the solution to that is to get all steps and then manually pick out each of the individual steps per difficulty in the order we want).